code/modules/recycling/disposal-construction.dm

```diff
@@ -137,10 +137,41 @@
 // wrench: (un)anchor
 // weldingtool: convert to real pipe
 
-/obj/structure/disposalconstruct/attackby(obj/item/I, mob/user, params)
+/obj/structure/disposalconstruct/wrench_act(mob/living/user, obj/item/I)
+	var/ispipe = is_pipe()
+	var/nicetype = get_nice_name()
+	if(anchored)
+		anchored = FALSE
+		if(ispipe)
+			level = 2
+			density = FALSE
+		else
+			density = TRUE
+		to_chat(user, "You detach the [nicetype] from the underfloor.")
+	else
+		anchored = TRUE
+		if(ispipe)
+			level = 1 // We don't want disposal bins to disappear under the floors
+			density = FALSE
+		else
+			density = TRUE // We don't want disposal bins or outlets to go density 0
+		to_chat(user, "You attach the [nicetype] to the underfloor.")
```

Comment:
Should these two be using some type of span class? Similar tool to chat messages are using `notice` as the default.

Reply:
They should, disposals code is just so dusty and old

Reply:
Well... I left you some reviews. 😄